### PR TITLE
5515 Adjust freeze encoder of hovernet

### LINKED ISSUE #5515: Adjust the encoder freeze function on HoverNet
In the original paper (https://arxiv.org/pdf/1812.06499.pdf), the freeze way is to freeze the whole encoder.

However, during their official implementation, the freeze way is different on the first residual block (see: https://github.com/vqdang/hover_net/blob/842964dc5d26ffe196d236d16fad16643a103850/models/hovernet/net_desc.py#L107)

We may need to adjust the freeze function to align with their implementation.

cc @KumoLiu @JHancox

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Signed-off-by: Yiheng Wang <vennw@nvidia.com>

Fixes #5515  .

### Description

This PR adjusts the freeze encoder function, and also fix the issue of un-trigger some of the unit tests.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/nets/hovernet.py [hunk 1]
```diff
@@ -423,6 +433,7 @@ class HoVerNet(nn.Module):
             repository, each user is responsible for checking the content of model/datasets and the applicable licenses
             and determining if suitable for the intended use. please check the following link for more details:
             https://github.com/vqdang/hover_net#data-format
+        freeze_encoder: whether to freeze the encoder of the network.
```
Comment (Nic-Ma) [inline monai/networks/nets/hovernet.py:436]: "whether to freeze the encoder of the network when executing the backpropagation."
What do you think?

Thanks.

Comment (yiheng-wang-nv) [reply monai/networks/nets/hovernet.py:436]: Hi @Nic-Ma , IMO, we may not need to bind "freeze" with "executing the backpropagation". I did not see this kind of description in other places.

Comment (Nic-Ma) [reply monai/networks/nets/hovernet.py:436]: OK, I see.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**